### PR TITLE
Replace input/raw_input with prompt_toolkit

### PR DESCRIPTION
**Pull request by wwj718**
[prompt_toolkit](https://github.com/jonathanslenders/python-prompt-toolkit) is powerful interactive command tool to interact with the user. 
-  Support for history
-  Reverse and forward incremental search.
-  Works well with Unicode double width characters. (Chinese input.)
-  Selecting text for copy/paste. (Both Emacs and Vi style.)
-  Auto suggestions. (Like fish shell.)
-  Advanced code completion.(maybe we will need it in the future)

### CODE REVIEW

File: wit/wit.py [hunk 1]
```diff
@@ -154,9 +157,11 @@ def interactive(self, context=None, max_steps=DEFAULT_MAX_STEPS):
             input_function = input
 
         session_id = uuid.uuid1()
+        history = InMemoryHistory()
         while True:
             try:
-                message = input_function(INTERACTIVE_PROMPT).rstrip()
+                #message = input_function(INTERACTIVE_PROMPT).rstrip()
```
Comment (patapizza) [inline wit/wit.py]: nit: rm

File: wit/wit.py [hunk 2]
```diff
@@ -154,9 +157,11 @@ def interactive(self, context=None, max_steps=DEFAULT_MAX_STEPS):
...
+                message = prompt(INTERACTIVE_PROMPT, history=history,mouse_support=True).rstrip()
```
Comment (patapizza) [inline wit/wit.py]: nit: space after ,

**Comment by wwj718** [comment]
Ok,let me try it

**Comment by wwj718** [comment]
@patapizza  👌

File: examples/quickstart.py [hunk 3]
```diff
@@ -1,6 +1,18 @@
+#!/usr/bin/env python
```
Comment (patapizza) [inline examples/quickstart.py]: revert this file

File: wit/wit.py [hunk 4]
```diff
@@ -37,6 +40,7 @@ def req(logger, access_token, meth, path, params, **kwargs):
         raise WitError('Wit responded with an error: ' + json['error'])
 
     logger.debug('%s %s %s', meth, full_url, json)
+    print(json) #todo:use logging
```
Comment (patapizza) [inline wit/wit.py]: revert

File: wit/wit.py [hunk 5]
```diff
@@ -128,6 +132,7 @@ def __run_actions(self, session_id, current_request, message, context, i,
             json['action'] = 'merge'
 
         if json['type'] == 'error':
+            print(json)
```
Comment (patapizza) [inline wit/wit.py]: revert

File: wit/wit.py [hunk 6]
```diff
@@ -204,9 +209,11 @@ def interactive(self, context=None, max_steps=DEFAULT_MAX_STEPS):
             input_function = input
 
         session_id = uuid.uuid1()
+        history = InMemoryHistory()
         while True:
             try:
-                message = input_function(INTERACTIVE_PROMPT).rstrip()
+                #message = input_function(INTERACTIVE_PROMPT).rstrip()
```
Comment (patapizza) [inline wit/wit.py]: remove

**Comment by wwj718** [comment]
@patapizza  👌

**Comment by patapizza** [comment]
Thanks!

**Pull request merged**